### PR TITLE
Garden Endpoint

### PR DESCRIPTION
**Pull request by kelshadarby**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling (Styling to new or existing files)
- [X] Testing
### Detailed Description
- Create garden endpoint
    - This endpoint returns all plants in your garden
### Why is this change required? What problem does it solve?
- This was created to serve as a new piece of functionality
### Was any previously written code changed?
No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,22 +67,51 @@ Step 3:
...
 oh-crop-be.herokuapp.com/api/v1/
 ```
 
 __All Plants__
+
 This endpoint returns all plant object in the database
 
 ```
-oh-crop-be.herokuapp.com/api/v1/plants
+Request:
```
Comment (mikez321) [inline README.md:84]: I like that we're keeping these endpoints up to date as we go!

File: alembic/versions/165f21d7f610_create_new_garden_table.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+"""Create New Garden Table
```
Comment (mikez321) [inline alembic/versions/165f21d7f610_create_new_garden_table.py:1]: Having a proper many to many relationship really made this less brain-bendy, AND if we we were to upscale the app that part of functionality is already built in now.  Glad we spent the time on this instead of trying to do it in some weird kinda way.

File: tests/test_endpoint.py [hunk 3]
```diff
@@ -65,16 +66,34 @@ def test_api_can_add_plant_to_garden(self):
...
+        db.session.commit()
+
+        res = self.client().post('/api/v1/garden?plant_id={}&plant_name=Ezekiel'.format(zeke.id))
+        res = self.client().post('/api/v1/garden?plant_id={}&plant_name=Dan'.format(dan.id))
+
+        res = self.client().get('/api/v1/garden')
+        self.assertEqual(res.status_code, 200)
+        self.assertIn("Ezekiel", str(res.data))
+        self.assertIn("Dan", str(res.data))
+        self.assertNotIn("Agatha", str(res.data))
```
Comment (mikez321) [inline tests/test_endpoint.py:95]: Testing will be a big 'come back to this' later.  I'm sure that there are some edge cases we should be testing for.

File: app/models.py [hunk 4]
```diff
@@ -24,11 +27,21 @@ def random_id():
...
     date_added = db.Column(db.DateTime, default=datetime.now())
     harvest_date = db.Column(db.DateTime)
+    
+    plant = relationship("Plant")
+
+class Garden(db.Model):
+    __tablename__ = 'gardens'
+    id = db.Column(db.Integer, primary_key=True)
+
+    plants = relationship("Plant", secondary="garden_plants")
```
Comment (mikez321) [inline app/models.py:46]: This was WAY easier than I thought it would be!  Very excited that we were able to get methods like garden.gardenplants to work... and now we can be all like... garden.gardenplants.plants[0].water_needs and get values that will help us calculate things!  Great work!

**Comment by mikez321** [comment]
It was a little of a hassle to create all those new tables, db migrations, a joins table etc when we didn't totally need it, but I think it really helped us visualize what we were doing.  And in the end (now that we're getting the hang of things) it wasn't even too bad to create proper relationships!
